server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java

```diff
@@ -197,22 +197,23 @@ private void validateArguments(CreationMessageEntry entry, MailboxSession sessio
     @VisibleForTesting void assertAttachmentsExist(CreationMessageEntry entry, MailboxSession session) throws AttachmentsNotFoundException, MailboxException {
         List<Attachment> attachments = entry.getValue().getAttachments();
         if (!attachments.isEmpty()) {
-            AttachmentMapper attachmentMapper = attachmentMapperFactory.getAttachmentMapper(session);
-            List<BlobId> notFounds = listAttachmentsNotFound(attachments, attachmentMapper);
+            List<BlobId> notFounds = listAttachmentsNotFound(attachments, session);
             if (!notFounds.isEmpty()) {
                 throw new AttachmentsNotFoundException(notFounds);
             }
         }
     }
 
-    private List<BlobId> listAttachmentsNotFound(List<Attachment> attachments, AttachmentMapper attachmentMapper) {
+    private List<BlobId> listAttachmentsNotFound(List<Attachment> attachments, MailboxSession session) throws MailboxException {
         return attachments.stream()
             .flatMap(attachment -> {
                 try {
-                    attachmentMapper.getAttachment(getAttachmentId(attachment));
+                    attachmentManager.getAttachment(getAttachmentId(attachment), session);
                     return Stream.of();
                 } catch (AttachmentNotFoundException e) {
                     return Stream.of(attachment.getBlobId());
+                } catch (MailboxException e) {
+                    throw Throwables.propagate(e);
```

Comment:
So now it's a RuntimeException, and then will not be catched by the main method (handleCreate).